ghost/collections/src/Entity.ts

```diff
@@ -0,0 +1,104 @@
+
```

Comment:
Here be dragons

Reply:
This would become a base for all future entities in the system, right? So, similarly to the `domain-events` package we'd probably want a `domain-entity` package?

Reply:
Yeah exactly, haven't split it out yet as it was just playing around with it!